### DISCUSSION TITLE
Image thumbnails causing fly.io free instance to run out of memory

### DISCUSSION THREAD

**Discussion by tigawanna** [pocketbase/pocketbase#1920] [Q&A] [answered]
I don't know if am the only one who noticed this but this cool feature that I use to load the thumbnails as placeholder images while the higher quality one load in the background cause the fly.io instance to run out of memory, there's always the option of adding  1GB in memory but was bringing this up in case it hadn't come to anyone else's attention

**Comment by ganigeorgiev** [comment, marked as answer]
Could you provide a minimal repo or at least a code sample how and what requests you are sending? Also what version of PocketBase you are using?

The free fly vm comes with very little memory (256MB) so there isn't really much we can do and in most cases the free plan is suitable for POC and small/low-traffic apps (_unless you allocate more resources_).

If you need a more beefy server you can also check the free Oracle Ampere offering - https://www.oracle.com/cloud/free/ (_I haven't personally tested it yet_).

**Reply by ganigeorgiev** [reply]
One possible workaround I forgot to mention is to try with the [`GOMEMLIMIT` env variable](https://pkg.go.dev/runtime@master#hdr-Environment_Variables) (eg. `ENV GOMEMLIMIT=200MiB`).

**Reply by tigawanna** [reply]
> Could you provide a minimal repo or at least a code sample how and what requests you are sending? Also what version of PocketBase you are using?
> 
Nothing complex just grabbing a list of items and grabbing the thumbnail and the actual image the thumbnail loads way faster so that our list has some sort of image before the actual image loads

```tsx
         {
        data && data?.items.map((land) => {
        const img_url = makeImageUrl('listings', land.id, land.images[0] as string);
        const alt_img_url = makeImageUrl('listings', land.id, land.images[0] as string) + '?thumb=100x100';

        return (
        <Link href={`/listings/${land.id}`}
            key={land.id}
            className=" w-full flex flex-col items-start shadow-lg border hover:shadow-lg  hover:shadow-slate-300 rounded-2xl ">
       <div className=" h-full w-full flex items-center justify-center relative">
        
        {land.status === "sold" ?
        <div className="w-full h-full absolute font-bold font-serif
        flex items-center justify-center  text-6xl  text-slate-50 bg-slate-500 bg-opacity-30">
        SOLD</div> : null}

        <GoodImage props={{ src: img_url as string, alt: land?.location,}}
            placeholderSrc={ alt_img_url}
            height={'200px'} width={'300px'}/>
        </div>
```

**Reply by tigawanna** [reply]
> One possible workaround I forgot to mention is to try with the [`GOMEMLIMIT` env variable](https://pkg.go.dev/runtime@master#hdr-Environment_Variables) (eg. `ENV GOMEMLIMIT=200MiB`).

this will do the trick
